src/univariate/continuous/laplace.jl

```diff
@@ -114,8 +114,27 @@ rand(rng::AbstractRNG, d::Laplace) =
 
 #### Fitting
 
-function fit_mle(::Type{<:Laplace}, x::Array)
-    xc = copy(x)
-    a = median!(xc)
-    Laplace(a, StatsBase.mad!(xc, center=a))
+function fit_mle(::Type{<:Laplace}, x::AbstractArray{T}) where {T <: Real}
+    μ = median(x)
+    return Laplace(μ, mean(abs.(x .- μ)))
```

Comment:
I would suggest:
```suggestion
    xc = similar(x)
    copyto!(xc, x)
    m = median!(xc)
    xc .= abs.(x .- m)
    return Laplace(m, mean(xc))
```
`similar` ensures (ideally) that `xc` is a mutable array.

Reply:
I made some changes which reduce memory allocation.

Reply:
As mentioned above, `median` creates an array internally and is not more memory efficient than this suggestion.

Reply:
I removed temporary allocations such as x .- m and abs.(...), which I believe allocate memory

Reply:
Yes, I agree, "median" requires an allocation.

Reply:
Thank you very much for recommending efficiency improvements.  My main issue at the moment is the current implementation is not a MLE.  I figured I would add a weighted MLE so the contribution had some additional marginal value.

Reply:
> I removed temporary allocations such as x .- m and abs.(...), which I believe allocate memory

The line
```julia
xc .= abs.(x .- m)
```
would not allocate memory but update `xc[i]` with `abs(x[i] - m)` for each index `i` (assuming both arrays use the same indices) without any temporary arrays.

>  My main issue at the moment is the current implementation is not a MLE. I figured I would add a weighted MLE so the contribution had some additional marginal value.

The bug should be fixed and it's great that you noted the issue and put together a PR. But actually it would be easier to approve the PR and make a bugfix release if it would not include any new features. I would prefer if bug fixes and PRs that add new features would be separated. I would like to hear the opinion of other contributors before approving and merging new features. In particular, IIRC it was mentioned previously that the general policy is to add not too many or too advanced estimation procedures.

So it would be great if you could minimize the changes in the PR and exclude the new estimation procedure, if possible.

Reply:
Very reasonable policy and I like your suggested implementation.   I will put your implementation in a PR for the bug fix.  Thanks.

Reply:
I created a new PR restricted to your implementation of the bug fix.  Thanks!